src/lib/ecs/query.ts

```diff
@@ -207,39 +205,25 @@ export class Query {
 	 *
 	 * @param callback The callback to run for each entity.
 	 */
-	public exitedQuery(callback: (entityId: EntityId) => boolean | void): void {
+	public *exitedQuery(): Generator<EntityId> {
 		for (const entityId of this.exited.dense) {
-			if (callback(entityId) === false) {
-				break;
-			}
+			yield entityId;
 		}
 
 		this.exited = new SparseSet();
 	}
 
 	/**
-	 * Runs a callback for each entity that matches the query.
-	 *
-	 * If the callback returns `false`, the iteration will stop, and no other
-	 * entities in this query will be iterated over.
-	 *
-	 * #### Usage Example:
-	 * ```ts
-	 * query.forEach((entity) => {
-	 * 	// ...
-	 * });
-	 * ```
-	 *
-	 * @param callback The callback to run for each entity.
+	 * @returns an array of all the entities that currently match the query.
```

Comment:
`entity IDs`*

Reply:
I basically use entityId and entities interchangeably in the docs anyway since they're the same thing. If you want me to change this one, I'd have to change a lot more in other places.